Trisha-tech/OnlineBookSales
# Fixed Search Function to Notify Users of Empty Input and No Matching Books

### Description
A clear and concise description of what the PR does.
- This PR does the following:
  Fixes the issue where no message was shown when a book search returned no results.
  Updates the search feature to display a notification when no book is found in the database.

### Related Issues
- "Fixes #423 "

### Changes
List the detailed changes made in this PR.
Added a pop-up message when the search returns no matching books.
Refactored the search component to handle empty result sets more efficiently.

### Testing Instructions
Detailed instructions on how to test the changes. Include any setup needed and specific test cases.
1. Pull this branch.
2. Run `npm install` to install dependencies.
3. Run `npm test` to execute the test suite.
4. Search for a non-existent book, and verify that a pop-up card or message is displayed, notifying the user that no results were found.

### Screenshots (if applicable)
https://github.com/user-attachments/assets/c80ca259-dbde-4dce-8d12-d19305314b42

### Additional Context
Any additional context or information that reviewers should be aware of.
- This PR is based on user feedback regarding the confusion when book search results are empty.

### Checklist
Make sure to check off all the items before submitting. Mark with [x] if done.
